### ISSUE TITLE
 [preamble] Not all code paths return a value

### ISSUE THREAD

**Issue by jakobo** [typed-graphql-builder/typed-graphql-builder#28]
Reproduction:
1. Run tgqlb on a schema
2. Attempt a compilation with the `tsc` command

Expected: Compiles w/o issues

Actual: `tsc` complains about the preamble output of `extractTextAndVars`

```
tsc --project ./services/worker/tsconfig.json --noEmit:
services/worker/src/__generated__/admin.ts:169:12 - error TS7030: Not all code paths return a value.

169   function extractTextAndVars(field: $Field<any, any, any> | $UnionSelection<any, any>) {
               ~~~~~~~~~~~~~~~~~~

Found 1 error in services/worker/src/__generated__/admin.ts:16
```

Proposed Fix: While _technically_ the if/elseif covers all permutation types, we should probably add an `else` clause that explicitly returns undefined if that's our intent. For now, it can be worked around by ts-ignoring the file.

**Comment by ilijaNL** [comment]
What tsconfig is used?

**Comment by jakobo** [comment]
Just a regular next.js config

```json
{
  "compilerOptions": {
    "baseUrl": "./",
    "rootDir": ".",
    "target": "es5",
    "lib": ["dom", "dom.iterable", "esnext"],
    "allowJs": true,
    "skipLibCheck": true,
    "strict": true,
    "forceConsistentCasingInFileNames": true,
    "noEmit": true,
    "esModuleInterop": true,
    "module": "esnext",
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "jsx": "preserve",
    "incremental": true,
    "paths": {
      "#~/*": ["./*"]
    }
  },
  "ts-node": {
    "files": true,
    "compilerOptions": {
      "sourceMap": true,
      "outDir": "dist",
      "strict": true,
      "esModuleInterop": true
    }
  },
  "include": ["next-env.d.ts", "additional.d.ts", "**/*.ts", "**/*.tsx"],
  "exclude": ["node_modules"]
}
```

I can also silence the error by setting `"noImplicitReturns": false`

**Comment by spion** [comment]
This should hopefully be fixed now in the latest published version (2.2.2). Let me know if that's not the case and I'll reopen

**Comment by jakobo** [comment]
Confirmed it's working now.
